frontend/src/store/credential.js

```diff
@@ -73,90 +76,164 @@ export const useCredentialStore = defineStore('credential', () => {
       set(state.secrets, [key], item)
     })
 
+    credentialsBindings?.forEach(item => {
+      const key = namespaceNameKey(item.metadata)
+      item.kind = 'CredentialsBinding' // ensure kind is set (might not be set if objects are retrieved using list call)
+      set(state.credentialsBindings, [key], item)
+    })
+
+    workloadIdentities?.forEach(item => {
+      const key = namespaceNameKey(item.metadata)
+      set(state.workloadIdentities, [key], item)
+    })
+
     quotas?.forEach(item => {
       const key = namespaceNameKey(item.metadata)
       set(state.quotas, [key], item)
     })
   }
 
-  const secretBindingList = computed(() => {
-    const secretBindings = Object.values(state.secretBindings)
-    return secretBindings.map(secretBinding => {
-      const secret = getSecret(secretBinding.secretRef)
-      Object.defineProperty(secretBinding, '_secret', {
-        value: secret,
-        configurable: true,
-        enumerable: false,
-      })
-
-      const quotas = (secretBinding.quotas || [])
-        .map(quota => get(state.quotas, namespaceNameKey(quota)))
-        .filter(Boolean)
-
-      Object.defineProperty(secretBinding, '_quotas', {
-        value: quotas,
-        configurable: true,
-        enumerable: false,
-      })
-
-      return secretBinding
-    })
-  })
-
-  const secretList = computed(() => {
-    return Object.values(state.secrets)
+  const cloudProviderBindingList = computed(() => {
+    const decorateBinding =
+     binding => {
+       const isSecretBinding = binding.kind === 'SecretBinding'
+       const isCredentialsBinding = binding.kind === 'CredentialsBinding'
+       const isInfrastructureBinding = cloudProfileStore.sortedProviderTypeList.includes(binding.provider?.type)
+       const isDnsBinding = gardenerExtensionStore.dnsProviderTypes.includes(binding.provider?.type)
+
+       if (isSecretBinding) {
+         const secret = getSecret(binding.secretRef)
+         Object.defineProperties(binding, {
+           _secret: {
+             value: secret,
+             configurable: true,
+           },
+           _secretName: {
+             value: binding.secretRef.name,
+             configurable: true,
+           },
+         })
+       }
+       if (isCredentialsBinding) {
+         if (binding.credentialsRef.kind === 'Secret') {
+           const secret = getSecret(binding.credentialsRef)
+           Object.defineProperties(binding, {
+             _secret: {
+               value: secret,
+               configurable: true,
+             },
+             _secretName: {
+               value: binding.credentialsRef.name,
+               configurable: true,
+             },
+           })
+         }
+         if (binding.credentialsRef.kind === 'WorkloadIdentity') {
```

Comment:
use e.g. `mdi-id-card` and key as icons, maybe also own columns for credential kind + type

Reply:
updated icons